packages/devtools_app/lib/src/screens/debugger/debugger_controller.dart

```diff
@@ -42,29 +43,32 @@ class DebuggerController extends DisposableController
     if (routerDelegate != null) {
       codeViewController.subscribeToRouterEvents(routerDelegate);
     }
-
-    evalService = EvalService(
-      isolateRef: isolateRef,
-      variables: variables,
-      frameForEval: () =>
-          _selectedStackFrame.value?.frame ??
-          _stackFramesWithLocation.value.safeFirst?.frame,
-      isPaused: isPaused,
+    addAutoDisposeListener(_selectedStackFrame, () => _updateCurrentFrame());
+    addAutoDisposeListener(
+      _stackFramesWithLocation,
+      () => _updateCurrentFrame(),
     );
 
-    setGlobal(EvalService, evalService);
-
     if (serviceManager.hasService) {
       _initialize();
     }
   }
 
-  final codeViewController = CodeViewController();
+  late final evalService = EvalService();
 
-  late final EvalService evalService;
+  final codeViewController = CodeViewController();
 
   bool _firstDebuggerScreenLoaded = false;
 
+  AppState get appState => serviceManager.appState;
```

Comment:
Why try to pretend that something from the serviceManager is part of the controller? I would just reference the serviceManager directly where this is used to make it clear that the DebuggerController is not needed for things it is not needed for. That will make it easier to detect the cases where we are now passing around the DebuggerController but don't need to.